dev/benchmarks/macrobenchmarks/README.md

```diff
@@ -49,3 +51,43 @@ flutter drive --profile test_driver/post_backdrop_filter_perf.dart
 Results should be in the file `build/post_backdrop_filter_perf.timeline_summary.json`.
 
 More detailed logs should be in `build/post_backdrop_filter_perf.timeline.json`.
+
+## Web benchmarks
+
+Web benchmarks are compiled from the same entrypoint in `lib/web_benchmarks.dart`.
+
+### How to write a web benchmark
+
+Create a new file for your benchmark under `lib/src/web`. See `bench_draw_rect.dart`
+as an example.
+
+Choose one of the two benchmark types:
+
+* A "raw benchmark" records performance metrics from direct interactions with
+  `dart:ui` with no framework. This kind of benchmark is good for benchmarking
+  low-level engine primitives, such as layer, picture, and semantics performance.
+* A "widget benchmark" records performance metrics using a widget. This kind of
+  benchmark is good for measuring the performance of widgets, often together with
+  engine work that widget-under-test incurs.
+
+For a raw benchmark implement `RawRecorder` (tip: you can start by copying
+`bench_draw_rect.dart`).
+
+For a widget benchmark implement `WidgetRecorder` (tip: you can start by copying
+`bench_simple_lazy_text_scroll.dart`).
+
+Pick a unique benchmark name and class name and add it to the `benchmarks` list
+in `lib/web_benchmarks.dart`.
```

Comment:
There are 3 names here that the developer has to pick: file name, class name, and `benchmarkName`. It would be nice to explain what purpose each name serves.

Reply:
I'd like to keep this document in the "how to" style so one can just follow the steps. It's ok if some explanation are missing, as long as they are not blocking adding a new benchmarks, or running/debugging an existing one.

Reply:
However, it's fair game to add as much explanation as possible in the dartdocs for classes in question. As you are refactoring these classes right now perhaps you can also add more docs? :)